pyanaconda/ui/gui/spokes/datetime_spoke.py

```diff
@@ -1062,51 +1063,71 @@ def _show_no_network_warning(self):
     def _show_no_ntp_server_warning(self):
         self.set_warning(_("You have no working NTP server configured"))
 
+    def _show_failed_ntp_start_warning(self):
+        self.set_warning(_("NTP service failed to start."))
+
+    def _show_failed_ntp_stop_warning(self):
+        self.set_warning(_("NTP service failed to stop."))
+
     def on_ntp_switched(self, switch, *args):
+        # The handler is blocked if we had to reset the switch.
+        # Unfortunately, handler_block_by_func does not work.
+        if self._block_on_ntp_switched:
+            self._block_on_ntp_switched = False
+            return
+
+        # Clear warning messages.
+        self.clear_info()
+
+        # Try to set NTP.
         if switch.get_active():
```

Comment:
I have a suggestion for an small improvement. What about moving the `if`-`else` parts to a separate methods. Something like:

```
if switch.get_active():
    self._try_to_enabled_ntp(...)
else:
    self._disable_ntp(...)
```

And the blocking could be done as a reaction on the output of these functions.